provider/pkg/provider/provider.go

```diff
@@ -1911,6 +1911,13 @@ func (k *kubeProvider) Create(
 		initialized = partialErr.Object()
 	}
 
+	// We need to delete the empty status field returned from the API server if we are in
+	// preview mode. Having the status field set will cause a panic during preview if the Pulumi
+	// program attempts to read the status field.
+	if req.GetPreview() {
+		unstructured.RemoveNestedField(initialized.Object, "status")
+	}
+
```

Comment:
I wonder, should the provider return _unknown_ rather than _undefined_.  When it is a preview, I do believe that the outputs may include unknowns, but I don't know the downstream effects.  WDYT?

Reply:
In a create dry run, the status field should really be undefined, so my preference would be to just unset the field.

Reply:
If `status` will be defined with some value after the create is finished, then I agree with @EronWright that it should be unknown during preview.